theano/sandbox/gpuarray/type.py

```diff
@@ -120,7 +190,7 @@ def filter_variable(self, other, allow_convert=True):
                                  str(self.broadcastable)))
```

Comment:
(A couple of lines above) Why only check allow_convert if the broadcastable patterns are different? That could change in the future, this is why we have factored that out into `convert_variable`.